src/Appium.Net/Appium/Service/AppiumServiceBuilder.cs

```diff
@@ -450,6 +478,9 @@ private string Args
             {
                 List<string> argList = new List<string>();
                 CheckAppiumJS();
+
+                argList.AddRange(NodeOptions);
```

Comment:
again, I'm not quite sure about escaping arguments having spaces. Looking at the code below this looks mandatory. Although, I would rather expect this behaviour to be automatic when we pass arguments array to the downstream command execution function.

I did not investigate the reason `Args` property is a string rather than a list, but this seems rather suspicious/fragile to me. @akinsolb what do you think?

Reply:
I agree. We should probably create a private method that formats/escapes arguments in this class

Reply:
should they be escaped in general? usually all exec-like function allow to pass arrays and know how to escape these without doing the routine on the client side. Maybe you know if the downstream function here supports that, so we could get rid of string representation and "manual" escaping at all?

Reply:
There's no magic like that here AFIK. They need to be escaped. The only reason why the manual ones exist below is because a bug was raised for not handling it.

Reply:
it looks like the standard Process class in sharp is less intellectual that I thought:
- http://csharptest.net/529/how-to-correctly-escape-command-line-arguments-in-c/index.html
- https://stackoverflow.com/questions/5510343/escape-command-line-arguments-in-c-sharp

Although, it always makes sense to use verified solutions rather than invent own wheels

Reply:
@sparerd Could you please add a separate method for arguments quoting (for example from the links above) and then add unit tests to verify that arguments containing spaces and/or quotes are escaped properly?

Reply:
Because this new method takes in arbitrary arguments that will be passed along, it seems unnecessary and perhaps even dangerous to try to escape them. If we miss an edge case and cause a bug, it could prevent valid inputs. In that case the user is stuck - they would need to wait for a bug fix or try to find a work around.

I've been in that situation before, and fighting with Windows command line escaping *and* a library's attempt to make escaping easier isn't fun. You then need to write the correctly escaped string, then modify it further to avoid the library escaping code. As a user, I would much rather not have that feature. Or at least have a way to disable it.

It makes a bit more sense having escaping built in for the other builder methods like `WithLogFile` or the port selection methods because you do not have direct control other the string representation of those arguments. With `WithNodeArguments` users must necessarily have full control over the input. This could be made more explicit by renaming to `WithRawNodeArguments` or something like that and add some notes in the method docs.

Reply:
this makes a lot of sense. 

The reason I touched this topic is that there was an issue is java client while passing complicated json-string (this is a very common and valid case if one wants to have default capabilities). This type of data is quite complicated and must be escaped properly. Usually manual escaping is quite a tricky task and unfortunately the most of the users are unable to solve it. So I would rather stick to an option where we have this behaviour under a flag. Let say by default the library tries to escape args, but in case your input data is too complicated then there is a possibility to provide a flag/use an alternative method to not escape custom args. What do you think about such approach?

Reply:
I can understand the desire to reduce friction and user support requests, but I'm very reluctant to add a feature (the auto-escaping) which I myself will avoid using, especially when providing args to Node is probably already a pretty niche capability. I'm worried that the escaping development will take more effort than I'm willing to put in.

At the time of this PR, most Node args seem to be flags or require small and well defined inputs. A few take paths, which are easy enough to handle. The only really scary ones (`--evaluate`) could fall pretty easily in the "dont do that" category and have other options which are easier to escape and maintain (for evaluate, just put the code in a JS file to execute).

Reply:
ok, then it would only be necessary to mention that in the method's docstring, that the args must be escaped manually if needed, otherwise I'm fine with the PR. Thanks for your work.

Reply:
Happy to merge @sparerd  when the xml comments provides notice to escape strings

Reply:
I've updated the arg comment - let me know if anything else should be added.